bazel/packaging/packaging.bzl

```diff
@@ -358,3 +358,154 @@ redpanda_package = rule(
+    for b in package_content.cc_binaries:
+        package_files.append({
+            "path": "bin",
+            "name": b.basename.removesuffix("_ld").removesuffix("_patched"),
+            "source": b.path,
+        })
+
+    for sl in package_content.shared_libraries:
+        package_files.append({
+            "path": "lib",
+            "name": sl.basename,
+            "source": sl.path,
+        })
+
+    return {
+        "package_dirs": ["bin", "lib"],
+        "directory_mode": True,
+        "owner": ctx.attr.owner,  # Default owner
+        "package_files": package_files,
+    }
+
+def _create_tarball_cfg(ctx, package_content):
+    package_files = []
+    for b in package_content.binaries:
+        package_files.append({
+            "path": "bin",
+            "name": b.basename,
+            "source": b.path,
+        })
+
+    # in tarball we put cc_binaries into libexec so they can be executed with a script that is created later
+    for b in package_content.cc_binaries:
+        package_files.append({
+            "path": "bin",
+            "name": b.basename.removesuffix("_ld").removesuffix("_patched"),
+            "source": b.path,
+        })
+
+    for sl in package_content.shared_libraries:
+        package_files.append({
+            "path": "lib",
+            "name": sl.basename,
+            "source": sl.path,
+        })
+
+    return {
+        "package_dirs": ["bin", "lib"],
+        "directory_mode": False,
+        "owner": ctx.attr.owner,  # Default owner
+        "package_files": package_files,
+    }
+
+def _native_pkg_impl(ctx):
+    use_dir = not ctx.attr.out.endswith(".tar.gz")
+    out = ctx.actions.declare_directory(ctx.attr.out) if use_dir else ctx.actions.declare_file(ctx.attr.out)
+    package_content = _prepapare_package_conent(ctx)
+    if use_dir:
+        cfg = _create_directory_cfg(ctx, package_content)
+    else:
+        cfg = _create_tarball_cfg(ctx, package_content)
+
+    # Create the configuration file for the packaging tool
+    cfg_file = ctx.actions.declare_file("%s.config.json" % ctx.attr.name)
+    ctx.actions.write(cfg_file, content = json.encode_indent(cfg))
+
+    inputs = [cfg_file] + package_content.shared_libraries + package_content.binaries + package_content.cc_binaries
+
+    # run the packaging tool
+    ctx.actions.run(
+        outputs = [out],
+        inputs = inputs,
+        tools = [ctx.executable._tool],
+        executable = ctx.executable._tool,
+        arguments = [
+            "-config",
+            cfg_file.path,
+            "-output",
+            out.path,
+        ],
+        mnemonic = "BuildingRedpandaPackage",
+        use_default_shell_env = False,
+    )
+    return [DefaultInfo(files = depset([out]))]
+
+native_package = rule(
+    implementation = _native_pkg_impl,
+    attrs = {
+        "cc_binaries": attr.label_list(),
+        "binaries": attr.label_list(),
```

Comment:
Why do we need both? Can we just unify? I guess we don't run patchelf on `binaries` right? For now let's assume YAGNI and we only will need this for binaries with shared libraries.